### PR TITLE
HDDS-10628. Display if safemode exit was via force exit command.

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
Safe mode rules are not calculated/updated after a safemode force exit, If a user looks at the SCM web UI after a force exit , it would give an impression that SCM is in safemode by looking at the rules not being satisfied.
Displaying if safe mode was forcefully exited will be helpful in  this case.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10628

## How was this patch tested?
Manually
Screenshot 2024-04-08 at 4 59 22 PM

### CODE REVIEW

**Comment by kerneltime** [comment]
cc @Tejaskriya

**Review by sumitagrawl — APPROVED** [review]
@sadanand48 LGTM

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java [hunk 1]
```diff
@@ -319,6 +323,14 @@ public void setPreCheckComplete(boolean newState) {
     this.preCheckComplete.set(newState);
   }
 
+  public boolean isSafeModeExitForceful() {
+    return forceExitSafeMode.get();
+  }
```
Comment (ayushtkn) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java:328]: the name of the variable is ``forceExitSafeMode`` the setter is also ``setForceExitSafeMode`` can we keep this method name in line: ``isForceExitSafeMode()``

**Review by Tejaskriya — APPROVED** [review]
LGTM +1

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java (hunk 1 above)
Comment (sadanand48) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java]: Done.

**Review by ayushtkn — APPROVED** [review]
LGTM

**Comment by adoroszlai** [comment]
Thanks @sadanand48 for the patch, @ayushtkn, @myskov, @sumitagrawl, @Tejaskriya for the review.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java [hunk 2]
```diff
@@ -319,6 +328,14 @@ public void setPreCheckComplete(boolean newState) {
     this.preCheckComplete.set(newState);
   }
 
+  public boolean getForceExitSafeMode() {
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java]: This is addressed in b4cf6aa241d3017f58d27d9fc114d8e9362005d5

**Pull request merged**